### PR TITLE
Added testcase for smtstate tool.

### PR DESCRIPTION
**Pull request by Pavithra1602**
This script tests smtstate tool for all smt states.

Signed-off-by: Pavithra <pavrampu@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by Pavithra1602** [comment]
[console_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/6939047/console_log.txt)

File: ras/smtstate.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
```
Comment (PraveenPenguin) [inline ras/smtstate.py:12]: Copyright signature please  

```#
# Copyright: 2021 IBM
```

File: ras/smtstate.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            self.cancel("Test case is supported only on RHEL and SLES")
+
+    def test(self):
+
+        process.system("ppc64_cpu --smt=on")
+        for i in [ "off", "on", 4, 2 ]:
+            for j in [ 2, 4, "on", "off" ]:
+                process.system("ppc64_cpu --smt=%s" % j)
+                smt_initial = process.system_output("ppc64_cpu --smt", shell=True)
+                process.system("smtstate --save")
```
Comment (PraveenPenguin) [inline ras/smtstate.py:49]: if this command failed is next steps make sense , i say we need to check and fail the test if command failed

File: ras/smtstate.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                process.system("ppc64_cpu --smt=%s" % i)
+                self.log.info("SMT level before load = %s" % process.system_output("ppc64_cpu --smt"))
+                process.system("smtstate --load")
```
Comment (PraveenPenguin) [inline ras/smtstate.py:52]: here also

File: ras/smtstate.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+class smtstate_tool(Test):
+
+    '''
+    smtstate tool allows to save the current smt value and restore it.  
+    '''
+
+    def setUp(self):
+
+        sm = SoftwareManager()
+        self.detected_distro = distro.detect()
```
Comment (PraveenPenguin) [inline ras/smtstate.py:29]: scope of variable can be method only

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**